cmd/genyaml/gen_kubectl_yaml.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+	"k8s.io/kubernetes/cmd/genutils"
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+type cmdOption struct {
+	Name         string
+	Shorthand    string `yaml:",omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty"`
+	Usage        string `yaml:",omitempty"`
+}
+
+type cmdDoc struct {
+	Name             string
+	Synopsis         string       `yaml:",omitempty"`
+	Description      string       `yaml:",omitempty"`
+	Options          []cmdOption  `yaml:",omitempty"`
+	InheritedOptions []cmdOption	`yaml:"inherited_options,omitempty"`
+	Example          string       `yaml:",omitempty"`
+	SeeAlso          []string     `yaml:"see_also,omitempty"`
+}
+
+func main() {
```

Comment:
Thanks for the feedback! Cobra at the moment doesn't seem to have a convenient GenYamlTree function, so I based this off of this old gen man pages script for the time being:

https://github.com/kubernetes/kubernetes/blob/643cb7a1c7499df4e569f4f0fbd3b18c0c4e63ce/cmd/genman/gen_kubectl_man.go

As a side note, since the time of that genman script's writing, cobra now has built-in man page generation (GenManTree function, see: https://github.com/spf13/hugo/issues/1369), so it might be possible to clean up that script and really easily extend man page generation to other kube commands (I can look into this if you'd like!).

Regarding the current code, maybe another option would be to suggest a PR adding a GenYamlTree function to the cobra repo, and if it does get approved, use that here in place of the messy custom logic? Though that might take some time.